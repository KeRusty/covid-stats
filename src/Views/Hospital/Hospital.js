import React, { useState, useEffect } from 'react';
import { Text, View, ScrollView } from 'react-native';

import Tests from '../../Components/PCRTestBox/PCRTestBox';
import Hospitals from '../../Components/Hospitals/Hospitals';

import styles from './Hospital-styles'

export default function Local() {

    const [loading, setLoading] = useState(true);
    const [data, setData] = useState();

    useEffect(() => {
        fetch('https://hpb.health.gov.lk/api/get-current-statistical')
            .then((response) => response.json())
            .then((json) => setData(json))
            .catch((error) => console.error(error))
            .finally(() => setLoading(false));
    }, []);

    const hospitalData = !loading ? data.data.hospital_data : [];

    return (
        <View style={styles.container}>

            {!loading &&
                <ScrollView contentContainerStyle={styles.scrollview}>

                    <View style={styles.heading}>

                        <Text style={styles.headingMainText}>COVID-19 Situation Report</Text>
                        <Text style={styles.headingSubText}>Last Update</Text>
                        <Text style={styles.headingTimeText}>{data.data.update_date_time}</Text>

                    </View>

                    {hospitalData.map((data, key) => (
                        <Hospitals
                            key={key}
                            content={data}
                        />
                    ))

                    }

                </ScrollView>
            }


        </View>
    );
}