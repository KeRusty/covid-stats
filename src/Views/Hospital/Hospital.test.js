import React from 'react';
import renderer from 'react-test-renderer';

import Hospital from './Hospital';

describe('<Hospital />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Hospital />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});