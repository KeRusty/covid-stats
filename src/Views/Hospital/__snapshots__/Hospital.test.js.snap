// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hospital /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
    }
  }
/>
`;
